consensus/src/pow/work.rs

```diff
@@ -49,11 +49,18 @@ pub fn check_proof_of_work(
 pub fn check_pow_consensus<H: BlockIndexHandle>(
     chain_config: &ChainConfig,
     header: &BlockHeader,
+    block_pow_data: &PoWData,
     pow_status: &PoWStatus,
     block_index_handle: &H,
 ) -> Result<(), ConsensusPoWError> {
     let work_required =
         calculate_work_required(chain_config, header, pow_status, block_index_handle)?;
+
+    // TODO: add test for a block with invalid target
+    if work_required != block_pow_data.bits() {
```

Comment:
Was also curious why do have to store target in the header. Couldn't it be calculated on the fly whenever required?

Reply:
See the comment on the PR

Reply:
sorry, missed the description